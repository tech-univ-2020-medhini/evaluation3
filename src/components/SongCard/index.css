.song-card{
    display: flex;
    flex-direction: column;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    transition: 0.3s;
    height: 20rem;
    width: 20rem;
    font-size: small;
}
.song:hover{
    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
}
img{
    max-height: 50%;
    max-width: 100%;
    margin: 1rem;
    border-radius: 50%;
  }