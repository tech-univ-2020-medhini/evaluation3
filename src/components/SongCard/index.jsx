import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import Button from '../Button';


function SongCard(props) {
  const {
    id, title, artists, imageLink,
  } = props;
  let artistStr;
  if (artists !== []) {
    artistStr = artists.join(' ');
  } else {
    artistStr = '';
  }

  const [likes, setLikes] = useState(0);
  useEffect(() => {
    const getLikes = async () => {
      const response = await axios.get(`/api/records/${id}/love`, {
        headers: {
          Authorization: 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1bmlxdWVfbmFtZSI6IjhmOGI3Y2E0LTE3OWEtNGNlMS04OTg0LWM0OTNiZWNlZTVjNyIsIm5iZiI6MTU4MTk5ODYzOCwiZXhwIjoxNTgyMDg1MDM4LCJpYXQiOjE1ODE5OTg2MzgsImlzcyI6InJlY29yZHMtc2hlbGYuY29tIn0.NPzsD41a6JZzywL7rCkgyqkBNWnxx5echV5a7MFGK7M',
        },
      });
      setLikes(response.data.data.count);
    };
    getLikes();
  }, []);
  const likesClick = () => {
    const patchLikes = async () => {
      axios.patch(`/api/records/${id}/love`, {
        count: likes + 1,
      }, {
        headers: {
          Authorization: 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1bmlxdWVfbmFtZSI6IjhmOGI3Y2E0LTE3OWEtNGNlMS04OTg0LWM0OTNiZWNlZTVjNyIsIm5iZiI6MTU4MTk5ODYzOCwiZXhwIjoxNTgyMDg1MDM4LCJpYXQiOjE1ODE5OTg2MzgsImlzcyI6InJlY29yZHMtc2hlbGYuY29tIn0.NPzsD41a6JZzywL7rCkgyqkBNWnxx5echV5a7MFGK7M',
        },
      });
      setLikes(likes + 1);
    };
    patchLikes();
  };

  return (
    <div className="SongCard">
      <img src={imageLink} alt="/" />
      <p>
        {title}
      </p>
      <p>
        {artistStr}
      </p>
      <Button text={likes} click={likesClick} />
    </div>

  );
}
SongCard.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  artists: PropTypes.arrayOf(PropTypes.string).isRequired,
  imageLink: PropTypes.string.isRequired,
};

export default SongCard;
