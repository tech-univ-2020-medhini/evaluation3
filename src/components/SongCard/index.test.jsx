import React from 'react';
import { render } from '@testing-library/react';
import SongCard from './index';


describe('the song cards', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<SongCard artists={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
