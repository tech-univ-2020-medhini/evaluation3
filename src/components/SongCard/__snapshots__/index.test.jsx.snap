// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the song cards should render correctly 1`] = `
<DocumentFragment>
  <div
    class="song-card"
  >
    <img
      alt="/"
    />
    <p />
    <p />
    <button
      class="Button"
      data-testid="test-btn"
      type="button"
    >
      0
    </button>
  </div>
</DocumentFragment>
`;
